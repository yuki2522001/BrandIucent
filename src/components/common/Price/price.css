.price__original,
.price__discount {
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
}

.price__original {
  color: var(--dark);
}

.price__discount {
  text-decoration-line: line-through;
  color: var(--bg-secondary);
}