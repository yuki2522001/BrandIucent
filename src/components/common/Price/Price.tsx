import React, { memo } from "react";
import "./price.css";

interface PriceProps {
  value?: number;
  type: string;
  currency?: string;
}

const Price: React.FC<PriceProps> = memo(({ value, type, currency }) => {
  let className = "";
  switch (type) {
    case "original":
      className = "price__original";
      break;
    case "discount":
      className = "price__discount";
      break;
    default:
      break;
  }
  return (
    <p data-testid="price" className={className}>
      <span className="price__unit">{currency}</span>{value}
    </p>
  );
});

export default Price;
