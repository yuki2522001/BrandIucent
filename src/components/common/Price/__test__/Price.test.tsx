import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Price from "../Price";

describe("Price component", () => {
  test("should render component Price type 'discount' ", () => {
    const { getByTestId } = render(<Price value={0} type="discount" />);
    expect(getByTestId("price")).toBeInTheDocument();
  });

  test("should render component Price type 'original' ", () => {
    const { getByTestId } = render(<Price value={0} type="original" />);
    expect(getByTestId("price")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Price value={0} type="discount" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
