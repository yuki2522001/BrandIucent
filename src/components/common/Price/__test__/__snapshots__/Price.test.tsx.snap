// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="price__discount"
    data-testid="price"
  >
    <span
      class="price__unit"
    />
    0
  </p>
</DocumentFragment>
`;
